a3update.py

```diff
@@ -169,27 +196,33 @@ def download_updated_workshop_mods():
 
 
 def check_workshop_mods():
-    mod_file = os.environ["WORKSHOP_MODS"]
-    if (mod_file == ''):
-        debug("WORKSHOP_MODS env variable not set, nothing to do.")
-        return
-    if (mod_file.startswith("http")):
-        with open("preset.html", "wb") as f:
-            f.write(request.urlopen(mod_file).read())
-        mod_file = "preset.html"
-    if not os.path.isfile(mod_file):
-        raise Exception('\n'.join([
-            'Unable to load WORKSHOP_MODS file!',
-            'The WORKSHOP_MODS file should be added to the volume in the arma directory'
-        ]))
-    with open(mod_file) as f:
-        html = f.read()
-        matches = re.finditer(WORKSHOP_ID_REGEX, html)
-        for _, match in enumerate(matches, start=1):
-            mod_id = match.group(1)
-            check_workshop_mod(mod_id)
-    download_updated_workshop_mods()
-    debug("Workshop mods ready\n{}".format(WORKSHOP_MODS))
+    for i in range(DOWNLOAD_ATTEMPTS):
+        debug("iteration {}".format(i))
+        mod_file = os.environ["WORKSHOP_MODS"]
+        if (mod_file == ''):
+            debug("WORKSHOP_MODS env variable not set, nothing to do.")
+            return
+        if (mod_file.startswith("http")):
+            with open("preset.html", "wb") as f:
+                f.write(request.urlopen(mod_file).read())
+            mod_file = "preset.html"
+        if not os.path.isfile(mod_file):
+            raise Exception('\n'.join([
+                'Unable to load WORKSHOP_MODS file!',
+                'The WORKSHOP_MODS file should be added to the volume in the arma directory'
+            ]))
+        with open(mod_file) as f:
+            html = f.read()
+            matches = re.finditer(WORKSHOP_ID_REGEX, html)
+            for _, match in enumerate(matches, start=1):
+                mod_id = match.group(1)
+                check_workshop_mod(mod_id)
+        download_updated_workshop_mods_script()
```

Comment:
Because mods are still bundled (and it's changed to all mods in this pr), if any download fails it will retry downloading every mod again from the start.

Reply:
That doesn't happen, it validates the currently downloaded mods. I've used it for a couple of months in my arma 3 community and we don't have to redownload everything every time.